### ISSUE TITLE
Unit tests fail with upper-case hostname

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#203]
Originally reported of Google Code with ID 147.

```
What software version are you running? Please provide the output of "gnt-
cluster --version" and "gnt-cluster version".

<b>What distribution are you using?</b>

<b>What steps will reproduce the problem?</b>
1. Have a host with hostname with an upper-case letter ('Garden' vs 'garden')
2. tar zxvf ganeti-2.4.0.tar.gz
3. make check-TESTS

<b>What is the expected output? What do you see instead?</b>

I expect the unit tests to pass.

<b>Please provide any additional information below.</b>

Running ganeti.config_unittest.py
EEEEEEEE.EEE..
======================================================================
ERROR: testAddGroupCanSkipUUIDCheck (__main__.TestConfigRunner)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test/ganeti.config_unittest.py", line 231, in testAddGroupCanSkipUUIDCheck
    cfg = self._get_object()
  File "./test/ganeti.config_unittest.py", line 66, in _get_object
    _getents=_StubGetEntResolver)
  File "/tmp/gntbuild.XU3lNo2p/ganeti/config.py", line 160, in __init__
    self._OpenConfig(accept_foreign)
  File "/tmp/gntbuild.XU3lNo2p/ganeti/config.py", line 1541, in _OpenConfig
    raise errors.ConfigurationError(msg)
ConfigurationError: The configuration denotes node garden as master, while my hostname is Garden; opening a foreign configuration is only possible in accept_foreign mode

======================================================================
ERROR: testAddGroupDoesNotPreserveFields (__main__.TestConfigRunner)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test/ganeti.config_unittest.py", line 222, in testAddGroupDoesNotPreserveFields
    cfg = self._get_object()
  File "./test/ganeti.config_unittest.py", line 66, in _get_object
    _getents=_StubGetEntResolver)
  File "/tmp/gntbuild.XU3lNo2p/ganeti/config.py", line 160, in __init__
    self._OpenConfig(accept_foreign)
  File "/tmp/gntbuild.XU3lNo2p/ganeti/config.py", line 1541, in _OpenConfig
    raise errors.ConfigurationError(msg)
ConfigurationError: The configuration denotes node garden as master, while my hostname is Garden; opening a foreign configuration is only possible in accept_foreign mode

======================================================================
ERROR: testAddGroupFillsFieldsIfMissing (__main__.TestConfigRunner)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test/ganeti.config_unittest.py", line 208, in testAddGroupFillsFieldsIfMissing
    cfg = self._get_object()
  File "./test/ganeti.config_unittest.py", line 66, in _get_object
    _getents=_StubGetEntResolver)
  File "/tmp/gntbuild.XU3lNo2p/ganeti/config.py", line 160, in __init__
    self._OpenConfig(accept_foreign)
  File "/tmp/gntbuild.XU3lNo2p/ganeti/config.py", line 1541, in _OpenConfig
    raise errors.ConfigurationError(msg)
ConfigurationError: The configuration denotes node garden as master, while my hostname is Garden; opening a foreign configuration is only possible in accept_foreign mode

======================================================================
ERROR: testAddGroupPreservesFields (__main__.TestConfigRunner)
----------------------------------------------------------------------
Traceback (most recent call last):  
  File "./test/ganeti.config_unittest.py", line 215, in testAddGroupPreservesFields
    cfg = self._get_object()
  File "./test/ganeti.config_unittest.py", line 66, in _get_object
    _getents=_StubGetEntResolver)   
  File "/tmp/gntbuild.XU3lNo2p/ganeti/config.py", line 160, in __init__
    self._OpenConfig(accept_foreign)
  File "/tmp/gntbuild.XU3lNo2p/ganeti/config.py", line 1541, in _OpenConfig
    raise errors.ConfigurationError(msg)
ConfigurationError: The configuration denotes node garden as master, while my hostname is Garden; opening a foreign configuration is only possible in accept_foreign mode

======================================================================
ERROR: Test instantiate config object
----------------------------------------------------------------------
Traceback (most recent call last):  
  File "./test/ganeti.config_unittest.py", line 111, in testEmpty
    self._get_object()
  File "./test/ganeti.config_unittest.py", line 66, in _get_object
    _getents=_StubGetEntResolver)   
  File "/tmp/gntbuild.XU3lNo2p/ganeti/config.py", line 160, in __init__
    self._OpenConfig(accept_foreign)
  File "/tmp/gntbuild.XU3lNo2p/ganeti/config.py", line 1541, in _OpenConfig
    raise errors.ConfigurationError(msg)
ConfigurationError: The configuration denotes node garden as master, while my hostname is Garden; opening a foreign configuration is only possible in accept_foreign mode

======================================================================
ERROR: testGetNdParamsDefault (__main__.TestConfigRunner)
----------------------------------------------------------------------
Traceback (most recent call last):  
  File "./test/ganeti.config_unittest.py", line 192, in testGetNdParamsDefault
    cfg = self._get_object()
  File "./test/ganeti.config_unittest.py", line 66, in _get_object
    _getents=_StubGetEntResolver)   
  File "/tmp/gntbuild.XU3lNo2p/ganeti/config.py", line 160, in __init__
    self._OpenConfig(accept_foreign)
  File "/tmp/gntbuild.XU3lNo2p/ganeti/config.py", line 1541, in _OpenConfig
    raise errors.ConfigurationError(msg)
ConfigurationError: The configuration denotes node garden as master, while my hostname is Garden; opening a foreign configuration is only possible in accept_foreign mode

======================================================================
ERROR: testGetNdParamsModifiedNode (__main__.TestConfigRunner)
----------------------------------------------------------------------
Traceback (most recent call last):  
  File "./test/ganeti.config_unittest.py", line 201, in testGetNdParamsModifiedNode
    cfg = self._get_object()
  File "./test/ganeti.config_unittest.py", line 66, in _get_object
    _getents=_StubGetEntResolver)   
  File "/tmp/gntbuild.XU3lNo2p/ganeti/config.py", line 160, in __init__
    self._OpenConfig(accept_foreign)
  File "/tmp/gntbuild.XU3lNo2p/ganeti/config.py", line 1541, in _OpenConfig
    raise errors.ConfigurationError(msg)
ConfigurationError: The configuration denotes node garden as master, while my hostname is Garden; opening a foreign configuration is only possible in accept_foreign mode

======================================================================
ERROR: Test initialize the config file
----------------------------------------------------------------------
Traceback (most recent call last):  
  File "./test/ganeti.config_unittest.py", line 115, in testInit
    cfg = self._get_object()
  File "./test/ganeti.config_unittest.py", line 66, in _get_object
    _getents=_StubGetEntResolver)   
  File "/tmp/gntbuild.XU3lNo2p/ganeti/config.py", line 160, in __init__
    self._OpenConfig(accept_foreign)
  File "/tmp/gntbuild.XU3lNo2p/ganeti/config.py", line 1541, in _OpenConfig
    raise errors.ConfigurationError(msg)
ConfigurationError: The configuration denotes node garden as master, while my hostname is Garden; opening a foreign configuration is only possible in accept_foreign mode

======================================================================
ERROR: Test updates on the cluster object
----------------------------------------------------------------------
Traceback (most recent call last):  
  File "./test/ganeti.config_unittest.py", line 121, in testUpdateCluster
    cfg = self._get_object()
  File "./test/ganeti.config_unittest.py", line 66, in _get_object
    _getents=_StubGetEntResolver)   
  File "/tmp/gntbuild.XU3lNo2p/ganeti/config.py", line 160, in __init__
    self._OpenConfig(accept_foreign)
  File "/tmp/gntbuild.XU3lNo2p/ganeti/config.py", line 1541, in _OpenConfig
    raise errors.ConfigurationError(msg)
ConfigurationError: The configuration denotes node garden as master, while my hostname is Garden; opening a foreign configuration is only possible in accept_foreign mode

======================================================================
ERROR: Test updates on one instance object
----------------------------------------------------------------------
Traceback (most recent call last):  
  File "./test/ganeti.config_unittest.py", line 155, in testUpdateInstance
    cfg = self._get_object()
  File "./test/ganeti.config_unittest.py", line 66, in _get_object
    _getents=_StubGetEntResolver)   
  File "/tmp/gntbuild.XU3lNo2p/ganeti/config.py", line 160, in __init__
    self._OpenConfig(accept_foreign)
  File "/tmp/gntbuild.XU3lNo2p/ganeti/config.py", line 1541, in _OpenConfig
    raise errors.ConfigurationError(msg)
ConfigurationError: The configuration denotes node garden as master, while my hostname is Garden; opening a foreign configuration is only possible in accept_foreign mode

======================================================================
ERROR: Test updates on one node object
----------------------------------------------------------------------
Traceback (most recent call last):  
  File "./test/ganeti.config_unittest.py", line 137, in testUpdateNode
    cfg = self._get_object()
  File "./test/ganeti.config_unittest.py", line 66, in _get_object
    _getents=_StubGetEntResolver)   
  File "/tmp/gntbuild.XU3lNo2p/ganeti/config.py", line 160, in __init__
    self._OpenConfig(accept_foreign)
  File "/tmp/gntbuild.XU3lNo2p/ganeti/config.py", line 1541, in _OpenConfig
    raise errors.ConfigurationError(msg)
ConfigurationError: The configuration denotes node garden as master, while my hostname is Garden; opening a foreign configuration is only possible in accept_foreign mode

----------------------------------------------------------------------
Ran 14 tests in 0.610s

FAILED (errors=11)
FAIL: test/ganeti.config_unittest.py

```
Originally added on 2011-03-07 19:53:52 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-07-19 13:14:34 +0000 UTC.

Added Labels: Type-Defect

**Comment by Ganeti-Issues-Migrator** [comment]
```
Easily reproduced using the following command:

$ GANETI_ROOTDIR=/tmp/FOO GANETI_HOSTNAME=FOO PYTHONPATH=. test/ganeti.config_unittest.py

```
Originally added on 2012-12-18 15:58:54 +0000 UTC.

Changed State: Accepted
Added to Milestone: Release2.7

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 5f30ea3
Author: Michael Hanselmann <hansmi@google.com>
Date:   Thu Dec 20 10:53:36 2012 +0100

    netutils: Normalize all FQDNs
    
    Addresses issue 147. Only the hostnames resolved using
    “netutils.Hostname($name)” were normalized, but those returned by the
    class method “GetSysName” weren't.
    
    Signed-off-by: Michael Hanselmann <hansmi@google.com>
    Reviewed-by: Helga Velroyen <helgav@google.com>

```
Originally added on 2012-12-20 10:38:15 +0000 UTC.

Mentioned Issue with GoogleCode ID 147 has been migrated to GitHub with ID #203.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking 2.7 bugs fixed, as 2.7.0~rc1 was released, and should be ready to be tested in production environments.
```
Originally added on 2013-05-09 05:42:48 +0000 UTC.

Changed State: Released
